src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/upgrade/upgrade.component.html

```diff
@@ -1,12 +1,18 @@
-<ng-container *ngIf="{upgradeInfo: upgradeInfo$ | async, error: upgradeInfoError$ } as upgrade">
-  <div class="row h-25 ms-1"
-       *ngIf="!upgrade.upgradeInfoError && upgrade.upgradeInfo as upgradeInfo; else checkUpgrade">
-    <ng-container *ngIf="healthData$ | async as healthData">
-      <div class="col-lg-3 h-50 d-flex flex-column border justify-content-center align-items-center">
+<div class="row h-25">
+  <ng-container *ngIf="healthData$ | async as healthData">
+    <cd-card class="col-sm-3 px-3 d-flex"
+             aria-label="Current Version"
```

Comment:
I agree Current version and then Upgrade Status are the only one's required. If some conditions are not met we can skip the upgrade button and add the conditions that are not satisfied.

Reply:
I had some progress but it felt like it need more discussion especially for `Service instances migration`. 
We can check if the daemons are migrated by comparing the daemon version with the current mgr version (which is what we consider as the cluster version). And as per cephadm implementation, you cannot upgrade any other daemons without first upgrading the mgr. So we are good there. I attached the screenshot below. There is also a tooltip which shows how many of the daemons are in new version.

I have removed those changes from the current PR. I'll open a tracker and track it for a different one after the discussions.

https://github.com/ceph/ceph/assets/71764184/396d06c2-80f2-491e-a679-48e4a43d61f5

Reply:
https://tracker.ceph.com/issues/62266

Reply:
Sorry for the mess: I mentioned the "service instances" thing because I feel like "daemon" is a very Linux-specific term, but as we talk about "daemons" all over the Dashboard, for consistency we should keep that until we choose "service instance" (perhaps too big :/) or something else that works better.